### PR TITLE
Clarify debugging examples with more comments

### PR DESCRIPTION
**Pull request by idelpivnitskiy**
Motivation:

Users unintentionally copy HTTP/2 configuration when they should debug
only HTTP/1.1 and still ask questions why trace logs are not visible.

Modifications:
- Clarify 2nd and 3rd argument of `enableWireLogging` and
`enableFrameLogging`;
- Clarify that `HttpProtocolConfigs.h2()` is required only for HTTP/2;
- Clarify how to `enableFrameLogging` in case of ALPN;

Result:

More comments for debugging examples.

### CODE REVIEW

File: servicetalk-examples/grpc/debugging/src/main/java/io/servicetalk/examples/grpc/debugging/DebuggingClient.java [hunk 1]
```diff
@@ -129,25 +129,36 @@ public static void main(String[] args) throws Exception {
...
 
                         /*
-                         * 4. Enables detailed logging of HTTP2 frames, but not frame contents.
-                         * Be sure to also enable the logger in your logging config file,
-                         * {@code log4j2.xml} for this example.
+                         * 4. Enables detailed logging of HTTP/2 frames.
+                         *
+                         * Be sure to also enable the TRACE logger in your logging config file (log4j2.xml for this
+                         * example) or rise the configured logging level (2nd argument) to INFO/WARNING to get
+                         * visibility without modifying the logger config.
```
Comment (bondolo) [inline servicetalk-examples/grpc/debugging/src/main/java/io/servicetalk/examples/grpc/debugging/DebuggingClient.java:157]:
```suggestion
                         * Be sure to also enable the {@code TRACE} logger in your logging config file (log4j2.xml
                         *  for this example) or raise the configured logging level (2nd argument) to
                         *  {@code INFO/WARNING} to get visibility without modifying the logger config.
```

File: servicetalk-examples/grpc/debugging/src/main/java/io/servicetalk/examples/grpc/debugging/DebuggingServer.java [hunk 2]
```diff
@@ -125,34 +125,43 @@ public final class DebuggingServer {
...
+                         * sensitive information into logs output. Be careful enabling data logging in production
+                         * environments.
+                         */
+                        .enableWireLogging("servicetalk-examples-wire-logger", TRACE, LOG_USER_DATA)
+
+                        /*
+                         * 4. Enables detailed logging of HTTP/2 frames.
+                         *
+                         * Be sure to also enable the TRACE logger in your logging config file (log4j2.xml for this
+                         * example) or rise the configured logging level (2nd argument) to INFO/WARNING to get
```
Comment (bondolo) [inline servicetalk-examples/grpc/debugging/src/main/java/io/servicetalk/examples/grpc/debugging/DebuggingServer.java:154]:
```suggestion
                         * example) or raise the configured logging level (2nd argument) to INFO/WARNING to get
```

File: servicetalk-examples/http/debugging/build.gradle [hunk 3]
```diff
@@ -21,5 +21,9 @@ dependencies {
   implementation project(":servicetalk-annotations")
   implementation project(":servicetalk-http-netty")
 
+  // This dependency brings self-signed TLS certificates for demonstration purposes.
+  // Users have to use their own certificates instead.
```
Comment (bondolo) [inline servicetalk-examples/http/debugging/build.gradle:25]:
```suggestion
  //  In real applications users have to provide their own certificates instead.
```

File: servicetalk-examples/http/debugging/build.gradle [hunk 4]
```diff
@@ -21,5 +21,9 @@ dependencies {
   implementation project(":servicetalk-annotations")
   implementation project(":servicetalk-http-netty")
 
+  // This dependency brings self-signed TLS certificates for demonstration purposes.
```
Comment (bondolo) [inline servicetalk-examples/http/debugging/build.gradle:24]:
```suggestion
  // This dependency provides sample logging configuration and self-signed TLS certificates for demonstration purposes.
```

File: servicetalk-examples/http/debugging/src/main/resources/log4j2.xml [hunk 5]
```diff
@@ -28,6 +28,7 @@
     <!-- Additional useful loggers of interest - include relevant loggers in your log4j2.xml file -->
```
Comment (bondolo) [inline servicetalk-examples/http/debugging/src/main/resources/log4j2.xml:28]: Is this file needed since the example also has a dependency on test resources? Perhaps also make this change in test resources log4j.xml as well.

**Review by bondolo — COMMENTED** [review]
Nice refinements and clarifications!

File: servicetalk-examples/grpc/debugging/src/main/java/io/servicetalk/examples/grpc/debugging/DebuggingClient.java (hunk 1 above)
Comment (bondolo) [reply servicetalk-examples/grpc/debugging/src/main/java/io/servicetalk/examples/grpc/debugging/DebuggingClient.java:157]: not a javadoc comment so I guess the `{@code}` suggestion can be ignored.

File: servicetalk-examples/http/debugging/src/main/resources/log4j2.xml (hunk 5 above)
Comment (idelpivnitskiy) [reply servicetalk-examples/http/debugging/src/main/resources/log4j2.xml:28]: I believe it's needed here. log4j.xml form test-resources is for tests purposes only, we don't want too much debug level logging there. In this example, it's useful to show what loggers they can enabled for extra visibility.

**Review by bondolo — APPROVED** [review]
